spec/entities/base_spec.rb

```diff
@@ -113,103 +187,181 @@ class Child < KalibroClient::Entities::Base; end
 
+    context 'with a successful response' do
+      before :each do
+        subject.class.stubs(:request).with('', anything, :post, '').
+          returns({"base" => {'id' => 42, 'errors' => []}})
+      end
+
+      context 'when it is not persisted' do
         it 'should make a request to save model with id and return true without errors' do
           expect(subject.save).to be(true)
+          expect(subject.id).to eq(42)
           expect(subject.kalibro_errors).to be_empty
         end
       end
 
-      context "when it returns with a kalibro processor error" do
+      context 'when it is persisted' do
         before :each do
-          KalibroClient::Entities::Base.
-            expects(:request).
-            with('', {base: {}}, :post, '').returns({"errors" => ["Name has already been taken"]})
+          subject.stubs(:persisted?).returns(true)
         end
 
-        it 'should make a request to save model returning false and a kalibro error' do
-          expect(subject.save).to be(false)
-          expect(subject.kalibro_errors[0]).to eq("Name has already been taken")
+        it 'is expected to call the update method'  do
+          subject.expects(:update).returns(true)
+          expect(subject.save).to eq(true)
         end
       end
     end
+  end
+
+  describe 'update' do
+    it_behaves_like 'persistence method', :update, :put
 
-    context 'when it is persisted' do
+    context 'with valid parameters' do
       before :each do
-        subject.persisted = true
-      end
+        id = 42
 
-      it 'is expected to call the update method'  do
-        subject.expects(:update)
+        subject.stubs(:id).returns(id)
+        KalibroClient::Entities::Base.expects(:request).with(':id', has_entry(id: id), :put, '').
+          returns({"base" => {'id' => id, 'errors' => []}})
+      end
 
-        subject.save
+      it 'is expected to return true' do
+        expect(subject.update).to eq(true)
       end
     end
   end
 
-  describe 'update' do
-    let!(:id) { 42 }
+  describe 'create' do
+    before :each do
+      subject.expects(:save)
+      KalibroClient::Entities::Base.
+        expects(:new).
+        with({}).
+        returns(subject)
+    end
 
-    context 'with valid parameters' do
+    it 'should instantiate and save the model' do
+      expect(KalibroClient::Entities::Base.create {}).to eq(subject)
+    end
+  end
+
+  describe 'find' do
+    context 'with an inexistent id' do
       before :each do
-        KalibroClient::Entities::Base.
-              expects(:request).
-              with(':id', {base: {}, id: id}, :put, '').returns({"base" => {'id' => id, 'kalibro_errors' => []}})
-        subject.expects(:id).returns(id)
-        subject.expects(:to_hash).returns({})
+        # We should remove this call in the future: there's no need to call exists before a find.
+        subject.class.stubs(:exists?).with(0).returns(false)
```

Comment:
That's the problem with `stubs`. https://github.com/mezuro/kalibro_client/commit/835180fab5d984fe02ea8753dfceeaa9ede41e87 has removed `exists` call but this mock was left behind since there was no failure.